src/typechecker/FStar.TypeChecker.Rel.fs

```diff
@@ -3386,6 +3386,16 @@ and solve_c (env:Env.env) (problem:problem<comp>) (wl:worklist) : solution =
           let stronger_t = stronger_t_opt |> must in
           let wl = extend_wl wl g_lift.deferred g_lift.deferred_to_tac g_lift.implicits in
 
+          if is_polymonadic &&
```

Comment:
Why is this is_polymonadic needed? Shouldn't this check occur for all lifts?

Reply:
For other lifts the check happens in TcUtil.lift_tf_layered_effect and TcUtil.get_mlift_for_subeff.